### PR TITLE
Even if jemalloc feature is used do not use it on msvc

### LINKED ISSUE #2233: cargo xtask lint fails on windows due to jemalloc-sys
We keep running into variations of this and sadly features are only opt-in on the command line. We also can't gate features on platform in `Cargo.toml` files which would keep us forwards compatible if `--all-features` is specified on the command line.

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes #2233

### CODE REVIEW

**Comment by matklad** [comment]
Great solution!

bors r+

Thanks!

**Comment by kjeremy** [comment]
I wish you could just ban a feature based on target or selectively disable it.

File: crates/ra_prof/Cargo.toml [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-once_cell = "1.0.1"
-itertools = "0.8.0"
-backtrace = "0.3.28"
-jemallocator = { version = "0.3.2", optional = true }
-jemalloc-ctl = { version = "0.3.2", optional = true }
-
-[features]
-jemalloc = [ "jemallocator", "jemalloc-ctl" ]
-cpu_profiler = []
+[package]
```
Comment (lnicola) [inline crates/ra_prof/Cargo.toml]: What happened here? Is git being weird about newlines?

Comment (kjeremy) [reply crates/ra_prof/Cargo.toml]: Hm... so I'm on windows and I just noticed that globally I have `core.autocrlf=true` but locally it's false... that seems wrong.

I think the correct fix is specifying everything as LF in .gitattributes?

Comment (lnicola) [reply crates/ra_prof/Cargo.toml]: No idea. A while ago I thought I understood how these worked, but then I realized I didn't.

**Pull request merged**